solr/core/src/java/org/apache/solr/core/DirectoryFactory.java

```diff
@@ -420,7 +420,7 @@ static DirectoryFactory loadDirectoryFactory(SolrConfig config, CoreContainer cc
     final DirectoryFactory dirFactory;
     if (info != null) {
       log.debug(info.className);
-      dirFactory = config.getResourceLoader().newInstance(info.className, DirectoryFactory.class);
+      dirFactory = config.getResourceLoader().newInstance(info, DirectoryFactory.class);
```

Comment:
Why?  FYI the ResourceLoader on the config is the same as that of the core, I believe.

Reply:
It's possible that we can share instances of SolrConfig between cores. We should always use SRL from core

Reply:
IndexSchema can be shared ("shareSchema" option) but SolrConfig at present cannot be.  Lets suppose it can be for the sake of argument -- wouldn't this be a reason to _not_ use SolrCore related things?  Otherwise another Collection using the same SolrConfig might theoretically refer to something it shouldn't.

Reply:
yeah, schema is shared. SolrConfig is not. Maybe it's safe